### ISSUE TITLE
Error during export of the project with Softbody, also errors in exported project. 

### ISSUE THREAD

**Issue by bartx** [godotengine/godot#82434]
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Godot v4.1.1.stable - Ubuntu 22.04.3 LTS 22.04 - Vulkan (Forward+) - dedicated AMD Radeon RX 6600 (RADV NAVI23) () - AMD Ryzen 7 5700G with Radeon Graphics (16 Threads)

### Issue description

Softbody attached to Rigidbody.
Errors during export. "scene/3d/node_3d.cpp:324 - Condition "!is_inside_tree()" is true. Returning: Transform3D())"
Softbody works correct in editor (follows Rigidbody), does not work correctly in Linux and Web exports (haven't tested other export profiles).

### Steps to reproduce

1. Project with one Rigidbody3D box and SoftBody attached to it by 4 points. No scripts. 
Project attached.

2. In editor it works ok (i.e. SoftBody follows falling RigidBody like a parachute)

3. During export I get 
 scene/3d/node_3d.cpp:324 - Condition "!is_inside_tree()" is true. Returning: Transform3D()
  scene/3d/node_3d.cpp:324 - Condition "!is_inside_tree()" is true. Returning: Transform3D()
  scene/3d/node_3d.cpp:324 - Condition "!is_inside_tree()" is true. Returning: Transform3D()
  scene/3d/node_3d.cpp:324 - Condition "!is_inside_tree()" is true. Returning: Transform3D()
  scene/3d/node_3d.cpp:324 - Condition "!is_inside_tree()" is true. Returning: Transform3D()
  scene/3d/node_3d.cpp:324 - Condition "!is_inside_tree()" is true. Returning: Transform3D()
  scene/3d/node_3d.cpp:324 - Condition "!is_inside_tree()" is true. Returning: Transform3D()
  scene/3d/node_3d.cpp:324 - Condition "!is_inside_tree()" is true. Returning: Transform3D()

4. On execution of exported project (both Linux and WASM): Softbody deforms correctly by gravity, but stays in one place (does not follow Rigidbody box). In console there are messages:

ERROR: Node3D node not defined in the pinned point, this is undefined behavior for SoftBody3D!
   at: _update_cache_pin_points_datas (scene/3d/soft_body_3d.cpp:719)
ERROR: Node3D node not defined in the pinned point, this is undefined behavior for SoftBody3D!
   at: _update_cache_pin_points_datas (scene/3d/soft_body_3d.cpp:719)
ERROR: Node3D node not defined in the pinned point, this is undefined behavior for SoftBody3D!
   at: _update_cache_pin_points_datas (scene/3d/soft_body_3d.cpp:719)
ERROR: Node3D node not defined in the pinned point, this is undefined behavior for SoftBody3D!

### Minimal reproduction project

[test.zip](https://github.com/godotengine/godot/files/12736993/test.zip)

**Comment by bartx** [comment]
Retested in 4.1.2 stable official, with the same results.

**Comment by pkmollman** [comment]
Can confirm, still seeing this in 4.1.3 stable.

**Comment by pkmollman** [comment]
I went ahead and set the attachments manually in the _ready() function and that seemed to fix it in the exported project.
Just pulled the lines from the .tscn file and wrapped in set()
That seems like a decent workaround for now.

```
extends SoftBody3D

func _ready():
	set("attachments/0/point_index", 29)
	set("attachments/0/spatial_attachment_path", NodePath("../Armature/Skeleton3D"))
	set("attachments/0/offset", Vector3(-0.2, 0.614552, 0.310236))
	set("attachments/1/point_index", 28)
	set("attachments/1/spatial_attachment_path", NodePath("../Armature/Skeleton3D"))
	set("attachments/1/offset", Vector3(-0.1, 0.614552, 0.310236))
	set("attachments/2/point_index", 27)
	set("attachments/2/spatial_attachment_path", NodePath("../Armature/Skeleton3D"))
	set("attachments/2/offset", Vector3(0, 0.614552, 0.310236))
	set("attachments/3/point_index", 26)
	set("attachments/3/spatial_attachment_path", NodePath("../Armature/Skeleton3D"))
	set("attachments/3/offset", Vector3(0.1, 0.614552, 0.310236))
	set("attachments/4/point_index", 25)
	set("attachments/4/spatial_attachment_path", NodePath("../Armature/Skeleton3D"))
	set("attachments/4/offset", Vector3(0.2, 0.614552, 0.310236))
```

**Comment by jonathanrivard** [comment]
Same issue in Godot 4.2.2, workaround from @pkmollman works

**Comment by lrasomattos** [comment]
Same issue when exporting for Android. (4.2.2 stable). SoftBody attachment doesn't work, until set manually on _ready
Same workaround.

**Comment by mihe** [comment]
I just had this reported to me in godot-jolt/godot-jolt#930 and it seems that this is a regression caused by godotengine/godot#71824.

In short, when scenes are exported they will first be instantiated ([here](https://github.com/godotengine/godot/blob/88f3b5f9d52f740b24fabfb8bc01b8b7026ba279/editor/export/editor_export_platform.cpp#L795)) in order to read its state. However, that instantiated scene is also immediately freed right after ([here](https://github.com/godotengine/godot/blob/88f3b5f9d52f740b24fabfb8bc01b8b7026ba279/editor/export/editor_export_platform.cpp#L825)) which means that deferred calls (like the one introduced in #71824) never have a chance to be flushed. Once they do get flushed (after the export is done) they'll act on orphaned nodes instead, causing a bunch of weird errors about `!is_inside_tree()`.

So pinning in general works thanks to `SoftBody3D::_set_property_pinned_points_indices` not being deferred, but `SoftBody3D::_set_property_pinned_points_attachment` with `"spatial_attachment_path"` is broken because it is deferred.

I'm not entirely sure if the correct thing is to flush the message queue as part of exporting, or if the thing that `call_deferred` in `SoftBody3D` was meant to fix should be fixed properly instead.

CC @rburing @PrecisionRender

**Comment by rburing** [comment]
@mihe I think https://github.com/godotengine/godot/pull/86310 should also fix this.

Ideally the deferred call introduced by https://github.com/godotengine/godot/pull/71824 would be removed, but I don't know how to solve the in-editor reparenting issue cleanly without it, given the order of operations performed by the editor when reparenting.

**Comment by mihe** [comment]
> I think #86310 should also fix this.

@rburing Ah, yeah, it might actually. I failed to realize that the instantiated scene is in fact never added to any tree, so I guess that would be covered by that check. I'll give it a try tomorrow.

**Comment by mihe** [comment]
@rburing I can confirm that #86310 fixes the issue, resulting the attachments working in the exported builds, and no errors being emitted during export.
